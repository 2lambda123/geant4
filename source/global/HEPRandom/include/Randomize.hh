// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: Randomize.hh,v 1.3 2000-08-02 10:48:21 gcosmo Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
#ifndef randomize_h
#define randomize_h 1

#include <CLHEP/Random/Randomize.h>

// Additional distributions ...
//
#include <CLHEP/Random/RandGaussQ.h>
#include <CLHEP/Random/RandGaussT.h>
#include <CLHEP/Random/RandPoissonQ.h>
#include <CLHEP/Random/RandPoissonT.h>
#include <CLHEP/Random/RandLandau.h>
#include <CLHEP/Random/RandBit.h>

#define G4UniformRand() HepRandom::getTheEngine()->flat()

#endif // randomize_h 
