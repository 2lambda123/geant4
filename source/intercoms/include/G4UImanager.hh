//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UImanager.hh,v 1.12 2001-10-02 00:32:06 asaim Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UImanager_h
#define G4UImanager_h 1

#include "globals.hh"
#include "g4rw/ctoken.h"
//#include "g4rw/tvordvec.h"
#include "g4std/vector"
#include "g4std/fstream"
#include "G4VStateDependent.hh"
#include "G4UIcommandStatus.hh"
class G4UIcommandTree;
class G4UIcommand;
class G4UIsession;
class G4UIcontrolMessenger;
class G4UnitsMessenger;
class G4UIaliasList;

// class description:
//
//  This is a singlton class which controls the command manipulation
// and the user interface(s). The constructor of this class MUST NOT 
// invoked by the user.
//

class G4UImanager : public G4VStateDependent
{
  public: // with description
      static G4UImanager * GetUIpointer();
      //  A static method to get the pointer to the only existing object
      // of this class.

  protected:
      G4UImanager();
  public:
      ~G4UImanager();
  private:
      G4UImanager(const G4UImanager &right);
      const G4UImanager & operator=(const G4UImanager &right);
      int operator==(const G4UImanager &right) const;
      int operator!=(const G4UImanager &right) const;

  public: // with description
      G4String GetCurrentValues(const char * aCommand);
      //  This method returns a string which represents the current value(s)
      // of the parameter(s) of the specified command. Null string will be
      // returned if the given command is not defined or the command does
      // not support the GetCurrentValues() method.
      void AddNewCommand(G4UIcommand * newCommand);
      //  This method register a new command.
      void RemoveCommand(G4UIcommand * aCommand);
      //  This command remove the registered command. After invokation of this
      // command, that particular command cannot be applied.
      void ExecuteMacroFile(G4String fileName);
      //  A macro file defined by the argument will be read by G4UIbatch object.
      G4int ApplyCommand(const char * aCommand);
      G4int ApplyCommand(G4String aCommand);
      //  These two methods are identical. A command (and parameter(s)) given
      // by the method's argument will be applied. Zero will be returned in 
      // case the command is successfully executed. Positive non-zero value
      // will be returned if the command couldn't be executed. The meaning of
      // this non-zero value is the following.
      //   The returned number : xyy
      //        x00 : G4CommandStatus.hh enumeration
      //         yy : the problematic parameter (first found)
      void StoreHistory(G4String fileName="G4history.macro");
      void StoreHistory(G4bool historySwitch,
             G4String fileName="G4history.macro");
      //  The executed commands will be stored in the defined file. If 
      // "historySwitch" is false, saving will be suspended.
      void ListCommands(G4String direc);
      //  All commands registored under the given directory will be listed to
      // G4cout.
      void SetAlias(const char * aliasLine);
      void SetAlias(G4String aliasLine);
      //  Define an alias. The first word of "aliasLine" string is the
      // alias name and the remaining word(s) is(are) string value
      // to be aliased.
      void RemoveAlias(const char * aliasName);
      void RemoveAlias(G4String aliasName);
      //  Remove the defined alias.
      void ListAlias();
      //  Print all aliases.


  public:
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4double initialValue,G4double finalValue,G4double stepSize);
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4int initialValue,G4int finalValue,G4int stepSize);

      virtual G4bool Notify(G4ApplicationState requestedState);
      //  This method is exclusively invoked by G4StateManager and the user
      // must not use this method.
 
  private:
      void PauseSession(G4String msg);
      void CreateMessenger();
      G4UIcommandTree* FindDirectory(const char* dirName);

  public:
  // following three methods will be removed quite soon.
      void Interact();
      void Interact(const char * promptCharacters);
      void Interact(G4String promptCharacters);

  private:
      static G4UImanager * fUImanager;
      static G4bool fUImanagerHasBeenKilled;
      G4UIcommandTree * treeTop;
      G4UIsession * session;
      G4UIcontrolMessenger * UImessenger;
      G4UnitsMessenger * UnitsMessenger;
      G4String savedParameters;
      G4UIcommand * savedCommand;
      G4int verboseLevel;
      G4std::ofstream historyFile;
      G4bool saveHistory;
      G4std::vector<G4String> histVec;
      G4UIaliasList* aliasList;
      
      G4bool pauseAtBeginOfEvent;
      G4bool pauseAtEndOfEvent;


  public: // with description
      G4String GetCurrentStringValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    int parameterNumber=1, G4bool reGet=true);
      G4String GetCurrentStringValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    const char * aParameterName, G4bool reGet=true);
      //  These six methods returns the current value of a parameter of the
      // given command. For the first three methods, the ordering number of
      // the parameter (1 is the first parameter) can be given, whereas,
      // other three methods can give the parameter name.
      //  If "reGet" is true, actual request of returning the current value
      // will be sent to the corresponding messenger, while, if it is false,
      // the value stored in G4Umanager will be used. The later case is valid
      // for the sequential invokation for the same command.

      inline void SetPauseAtBeginOfEvent(G4bool vl)
      { pauseAtBeginOfEvent = vl; }
      inline G4bool GetPauseAtBeginOfEvent() const
      { return pauseAtBeginOfEvent; }
      inline void SetPauseAtEndOfEvent(G4bool vl)
      { pauseAtEndOfEvent = vl; }
      inline G4bool GetPauseAtEndOfEvent() const
      { return pauseAtEndOfEvent; }
      //  If the boolean flags are true, Pause() method of G4StateManager is invoked
      // at the very begining (before generating a G4Event object) or at the end of
      // each event. So that, in case a (G)UI session is defined, the user can interact.


  public:
      inline G4UIcommandTree * GetTree() const
      { return treeTop; }
      inline G4UIsession * GetSession() const
      { return session; }
  public: // with description
      inline void SetSession(G4UIsession *const value)
      { session = value; }
      //  This method defines the active (G)UI session.
     void SetCoutDestination(G4UIsession *const value);
     //  This method defines the destination of G4cout/G4cerr stream.
     // For usual cases, this method will be invoked by a concrete
     // (G)UI session class object and thus the user needs not to invoke
     // this method by him(her)self.

  public:
      inline void SetVerboseLevel(G4int val)
      { verboseLevel = val; }
      inline G4int GetVerboseLevel() const
      { return verboseLevel; }
      inline G4int GetNumberOfHistory() const
      { return histVec.size(); }
      inline G4String GetPreviousCommand(G4int i) const
      { 
        G4String st;
        if(i>=0 && i<G4int(histVec.size()))
        { st = histVec[i]; }
        return st;
      }

  // Old methods kept for backward compatibility
  //    inline G4UIcommandTree * GetTree() const
  //    { return treeTop; };
  //    inline G4UIsession * GetSession() const
  //    { return session; };
  //    inline void SetSession(G4UIsession *const  value)
  //    { session = value; };

};

#endif

