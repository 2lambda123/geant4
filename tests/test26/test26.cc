//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: test26.cc,v 1.5 2003-03-26 17:29:30 vnivanch Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
/////////////////////////////////////////////////////////////////////////
//
// test26: Cut per region physics
//
// Created: 31.01.03 V.Ivanchenko
//
// Modified:
//
////////////////////////////////////////////////////////////////////////
//

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo...... 

#include "G4RunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "G4UItcsh.hh"
#include "Randomize.hh"


#include "Tst26DetectorConstruction.hh"
#include "Tst26PhysicsList.hh"
#include "Tst26PrimaryGeneratorAction.hh"
#include "Tst26RunAction.hh"
#include "Tst26EventAction.hh"
#include "Tst26TrackingAction.hh"
#include "Tst26SteppingAction.hh"
#include "Tst26SteppingVerbose.hh"


#ifdef G4VIS_USE
#include "Tst26VisManager.hh"
#endif

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......

int main(int argc,char** argv) {

  //choose the Random engine
  HepRandom::setTheEngine(new RanecuEngine);
  
  //my Verbose output class
  Tst26SteppingVerbose* stepvb = new Tst26SteppingVerbose();
  G4VSteppingVerbose::SetInstance(stepvb);

  // Construct the default run manager
  G4RunManager * runManager = new G4RunManager();

  // set mandatory initialization classes
  Tst26DetectorConstruction* det = new Tst26DetectorConstruction();
  runManager->SetUserInitialization(det);
  runManager->SetUserInitialization(new Tst26PhysicsList());

#ifdef G4VIS_USE
  // Visualization, if you choose to have it!
  G4VisManager* visManager = new Tst26VisManager();
  visManager->Initialize();
#endif

  Tst26PrimaryGeneratorAction* primary = new Tst26PrimaryGeneratorAction(det);
  runManager->SetUserAction(primary);

  // set user action classes
  Tst26RunAction* RunAct = new Tst26RunAction(primary);
  runManager->SetUserAction(RunAct);
  Tst26EventAction* EvtAct = new Tst26EventAction   (RunAct);
  runManager->SetUserAction(EvtAct);
  runManager->SetUserAction(new Tst26TrackingAction(RunAct));
  runManager->SetUserAction(new Tst26SteppingAction(EvtAct));

  // get the pointer to the User Interface manager
  G4UImanager* UI = G4UImanager::GetUIpointer();

  if (argc==1)   // Define UI terminal for interactive mode.
    {
      G4UIsession * session = 0;
#ifdef G4UI_USE_TCSH
      session = new G4UIterminal(new G4UItcsh);
#else
      session = new G4UIterminal();
#endif

      UI->ApplyCommand("/control/execute vis.mac");
      session->SessionStart();
      delete session;
    }
  else           // Batch mode
    {
      G4String command = "/control/execute ";
      G4String fileName = argv[1];
      UI->ApplyCommand(command+fileName);
    }

#ifdef G4VIS_USE
  delete visManager;
#endif
  delete runManager;
  delete stepvb;

  return 0;
}

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......

