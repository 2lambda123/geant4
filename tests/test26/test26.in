# $Id: test26.in,v 1.3 2003-02-07 15:33:02 vnivanch Exp $
#
# Macro file for "test26.cc" 
# V.Ivanchenko
#
#
/control/verbose 2
/run/verbose 1
#/tracking/verbose 1
#
#  select EM PhysicsList 
#
#/testem/phys/addPhysics  model
/testem/phys/addPhysics  standard
#
/testem/phys/WorldCuts  5.0 mm
/testem/phys/VertexCuts 0.01 mm
/testem/phys/MuonCuts  10.0 mm
/run/initialize
#
/gun/particle e-
/gun/energy 1 GeV
/run/beamOn 100
#
/gun/particle mu-
/gun/energy 1 GeV
/run/beamOn 100
#
/testem/phys/VertexCuts 1 mm
/run/beamOn 100
#
/gun/particle e-
/run/beamOn 100


