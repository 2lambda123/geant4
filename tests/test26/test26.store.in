# $Id: test26.store.in,v 1.1 2003-04-26 20:10:07 vnivanch Exp $
#
# Macro file for "test26.cc"
# V.Ivanchenko
#
#
/control/verbose 2
/run/verbose 1
/tracking/verbose 0
#
#  select EM PhysicsList
#
#/testem/phys/addPhysics  model
/testem/phys/addPhysics  standard
#
/testem/phys/WorldCuts  5.0 mm
/testem/phys/VertexCuts 0.01 mm
/testem/phys/MuonCuts  10.0 mm
/run/initialize
#
/gun/particle mu-
/gun/energy 1 GeV
/run/beamOn 100
/run/particle/storePhysicsTable physdata
#
/gun/particle pi+
/gun/energy 1 GeV
/run/beamOn 100
#
/gun/particle ion
/gun/ion      6 12 6 0
/gun/energy 1 GeV
/run/beamOn 100
#


