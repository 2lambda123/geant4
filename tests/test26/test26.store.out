**********************************************
 Geant4 version $Name: not supported by cvs2svn $
                                (30-Apr-2003)
             Copyright : Geant4 Collaboration
**********************************************
G4Material: Mean excitation energy is changed for Water Iold= 70.8926eV; Inew= 75 eV;
G4Material: Mean excitation energy is changed for CsI Iold= 491.166eV; Inew= 415.689 eV;

***** Table : Nb of materials = 11 *****

 Material: Air            density:  1.290 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 285.161 m
   --->  Element: NitrogenN   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %

 Material: Water            density:  1.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  36.092 cm
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material: lAr            density:  1.390 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  14.065 cm
   --->  Element: lAr    Z = 18.0   N =  40.0   A =  39.95 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Al            density:  2.700 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.893 cm
   --->  Element: Al    Z = 13.0   N =  27.0   A =  26.98 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Si            density:  2.330 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   9.368 cm
   --->  Element: Si    Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Fe            density:  7.870 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.759 cm
   --->  Element: Fe    Z = 26.0   N =  55.9   A =  55.85 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: CsI        CsI   density:  4.510 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.860 cm
   --->  Element: CesiumCs   Z = 55.0   N = 132.9   A = 132.91 g/mole  fractionMass:  51.15 %  Abundance  50.00 %
   --->  Element: IodideI   Z = 53.0   N = 126.9   A = 126.90 g/mole  fractionMass:  48.85 %  Abundance  50.00 %

 Material: BGO            density:  7.100 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.123 cm
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  15.41 %  Abundance  63.16 %
   --->  Element: GermaniumGe   Z = 32.0   N =  72.6   A =  72.59 g/mole  fractionMass:  17.48 %  Abundance  15.79 %
   --->  Element: BismuthBi   Z = 83.0   N = 209.0   A = 208.98 g/mole  fractionMass:  67.10 %  Abundance  21.05 %

 Material: PbWO4            density:  8.280 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.924 mm
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  14.07 %  Abundance  66.67 %
   --->  Element: LeadPb   Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass:  45.53 %  Abundance  16.67 %
   --->  Element: TungstenW   Z = 74.0   N = 183.8   A = 183.84 g/mole  fractionMass:  40.40 %  Abundance  16.67 %

 Material: Tungsten            density: 19.300 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   3.504 mm
   --->  Element: TungstenW   Z = 74.0   N = 183.8   A = 183.84 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Lead            density: 11.350 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   5.612 mm
   --->  Element: LeadPb   Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass: 100.00 %  Abundance 100.00 %


Visualization Manager instantiating...
Visualization Manager initialising...
Registering graphics systems...

You have successfully chosen to use the following graphics systems.
Current available graphics systems are:
  ASCIITree (ATree)
  DAWNFILE (DAWNFILE)
  GAGTree (GAGTree)
  G4HepRepFile (HepRepFile)
  G4HepRep (HepRepXML)
  RayTracer (RayTracer)
  VRML1FILE (VRML1FILE)
  VRML2FILE (VRML2FILE)
/run/verbose 1
/tracking/verbose 0
#
#  select EM PhysicsList
#
#/testem/phys/addPhysics  model
/testem/phys/addPhysics  standard
#
/testem/phys/WorldCuts  5.0 mm
/testem/phys/VertexCuts 0.01 mm
/testem/phys/MuonCuts  10.0 mm
/run/initialize
Ecal is 20 cm of PbWO4
Absorber is 2 mm of Al
Vertex is 10 cm of 3 layers of Si of 100 mum npads= 1999

phot:  Total cross sections from Sandia parametrisation. 
Tst26PhysicsList::SetCuts
#
/gun/particle mu-
/gun/energy 1 GeV
/run/beamOn 100

compt:  Total cross sections from a parametrisation. Good description from 10 KeV to (100/Z) GeV. 
       Scattered gamma energy according Klein-Nishina.
        PhysicsTables from 1 keV to 100 GeV in 80 bins. 

conv:  Total cross sections from a parametrisation. Good description from 1.5 MeV to 100 GeV for all Z. 
        e+e- energies according Bethe-Heitler
        PhysicsTables from 1.022 MeV to 100 GeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV to 100 TeV in 100 bins. 

eIoni:  delta cross sections from Moller+Bhabha. Good description from 1 KeV to 100 GeV.
        delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 100 TeV in 100 bins.
        Step function: finalRange(mm)= 1,  dRoverRange= 0.2

eBrem:  Total cross sections from a NEW parametrisation based on the EEDL data library. 
 Good description from 1 KeV to 100 GeV.
        log scale extrapolation above 100 GeV 
        Gamma energy sampled from a parametrised formula.
        PhysicsTables from 1 keV to 100 TeV in 100 bins. 

annihil:  Total cross section from Heilter formula(annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV to 100 TeV in 100 bins. 

hIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 100 TeV in 100 bins. 
        Step function: finalRange(mm)= 1,  dRoverRange= 0.2

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV to 100 TeV in 100 bins. 

MuIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

MuBrems:  theoretical cross section 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

MuPairProd:  theoretical cross sections 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 5 mm    e- 5 mm    e+ 5 mm
 Energy thresholds :  gamma 990 eV    e- 3.2397 keV    e+ 3.20006 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : PbWO4
 Range cuts        :  gamma 5 mm    e- 5 mm    e+ 5 mm
 Energy thresholds :  gamma 137.419 keV    e- 5.33819 MeV    e+ 4.9581 MeV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 2     used in the geometry : Yes     recalculation needed : No 
 Material : Al
 Range cuts        :  gamma 5 mm    e- 5 mm    e+ 5 mm
 Energy thresholds :  gamma 14.6203 keV    e- 2.31122 MeV    e+ 2.14666 MeV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 3     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 10 mum    e- 10 mum    e+ 10 mum
 Energy thresholds :  gamma 990 eV    e- 990 eV    e+ 990 eV
 Region(s) which use this couple : 
    VertexDetector

Index : 4     used in the geometry : Yes     recalculation needed : No 
 Material : Al
 Range cuts        :  gamma 10 mum    e- 10 mum    e+ 10 mum
 Energy thresholds :  gamma 990 eV    e- 33.8876 keV    e+ 33.4729 keV
 Region(s) which use this couple : 
    VertexDetector

Index : 5     used in the geometry : Yes     recalculation needed : No 
 Material : Si
 Range cuts        :  gamma 10 mum    e- 10 mum    e+ 10 mum
 Energy thresholds :  gamma 990 eV    e- 31.4747 keV    e+ 31.0896 keV
 Region(s) which use this couple : 
    VertexDetector

Index : 6     used in the geometry : Yes     recalculation needed : No 
 Material : Fe
 Range cuts        :  gamma 1 cm    e- 1 cm    e+ 1 cm
 Energy thresholds :  gamma 60.4019 keV    e- 12.9518 MeV    e+ 12.0297 MeV
 Region(s) which use this couple : 
    MuonDetector

Index : 7     used in the geometry : Yes     recalculation needed : No 
 Material : Al
 Range cuts        :  gamma 1 cm    e- 1 cm    e+ 1 cm
 Energy thresholds :  gamma 20.4949 keV    e- 4.54874 MeV    e+ 4.2772 MeV
 Region(s) which use this couple : 
    MuonDetector

====================================================================

### Run 0 start  beam E= 1000 MeV

--------- Ranecu engine status ---------
 Initial seed (index) = 0
 Current couple of seeds = 9876, 54321
----------------------------------------
Start Run processing.

---> Begin of Event: 0
Run terminated.
Run Summary
  Number of events processed : 100
  User=0.42s Real=0.44s Sys=0.01s

============   SUMMARY   ============

   100 Events were generated
   Incident particle mu- Ekin(meV) = 1000.00000

    Average number of secondaries

             World         Vertex Detector       Muon Detector

  Gammas:    7.82000        0.01000        1.89000
  Electrons: 14.80000        3.90000        6.46000
  Positrons:  0.33000        0.00000        6.46000

    Normalised energy deposition in the calorimeter

  E1 = 0.22323 +- 0.00158 RMS(E1) = 0.01582 +- 0.00158
  E9 = 0.22421 +- 0.00164 RMS(E9) = 0.01640 +- 0.00164
  E25= 0.22439 +- 0.00166 RMS(E25)= 0.01657 +- 0.00166

    Energy deposition in the absorbers

  Eabs1(MeV)= 0.79090 +- 0.02231 RMS(Eabs1)= 0.22308 +- 0.02231
  Eabs2(MeV)= 0.87151 +- 0.03413 RMS(Eabs2)= 0.34131 +- 0.03413
  Eabs3(MeV)= 0.86293 +- 0.04180 RMS(Eabs3)= 0.41803 +- 0.04180
  Eabs4(MeV)= 0.91967 +- 0.05368 RMS(Eabs4)= 0.53683 +- 0.05368

    Average number of hits in the vertex detector

  Nhit= 3.23000 +- 0.09680

--------- Ranecu engine status ---------
 Initial seed (index) = 0
 Current couple of seeds = 1798246724, 417411744
----------------------------------------
/run/particle/storePhysicsTable physdata
msc for GenericIon: Success to store the PhysicsTables in physdata
hIoni for GenericIon: Success to store the PhysicsTables in physdata
msc for He3: Success to store the PhysicsTables in physdata
hIoni for He3: Success to store the PhysicsTables in physdata
msc for alpha: Success to store the PhysicsTables in physdata
hIoni for alpha: Success to store the PhysicsTables in physdata
msc for anti_proton: Success to store the PhysicsTables in physdata
hIoni for anti_proton: Success to store the PhysicsTables in physdata
msc for deuteron: Success to store the PhysicsTables in physdata
hIoni for deuteron: Success to store the PhysicsTables in physdata
msc for e+: Success to store the PhysicsTables in physdata
eIoni for e+: Success to store the PhysicsTables in physdata
eBrem for e+: Success to store the PhysicsTables in physdata
annihil for e+: Success to store the PhysicsTables in physdata
msc for e-: Success to store the PhysicsTables in physdata
eIoni for e-: Success to store the PhysicsTables in physdata
eBrem for e-: Success to store the PhysicsTables in physdata
compt for gamma: Success to store the PhysicsTables in physdata
conv for gamma: Success to store the PhysicsTables in physdata
msc for kaon+: Success to store the PhysicsTables in physdata
hIoni for kaon+: Success to store the PhysicsTables in physdata
msc for kaon-: Success to store the PhysicsTables in physdata
hIoni for kaon-: Success to store the PhysicsTables in physdata
msc for mu+: Success to store the PhysicsTables in physdata
MuIoni for mu+: Success to store the PhysicsTables in physdata
MuBrems for mu+: Success to store the PhysicsTables in physdata
MuPairProdfor mu+: Success to store the PhysicsTables in physdata
msc for mu-: Success to store the PhysicsTables in physdata
MuIoni for mu-: Success to store the PhysicsTables in physdata
MuBrems for mu-: Success to store the PhysicsTables in physdata
MuPairProdfor mu-: Success to store the PhysicsTables in physdata
msc for pi+: Success to store the PhysicsTables in physdata
hIoni for pi+: Success to store the PhysicsTables in physdata
msc for pi-: Success to store the PhysicsTables in physdata
hIoni for pi-: Success to store the PhysicsTables in physdata
msc for proton: Success to store the PhysicsTables in physdata
hIoni for proton: Success to store the PhysicsTables in physdata
msc for triton: Success to store the PhysicsTables in physdata
hIoni for triton: Success to store the PhysicsTables in physdata
#
/gun/particle pi+
/gun/energy 1 GeV
/run/beamOn 100
### Run 1 start  beam E= 1000.00000 MeV

--------- Ranecu engine status ---------
 Initial seed (index) = 0
 Current couple of seeds = 1798246724, 417411744
----------------------------------------
Start Run processing.

---> Begin of Event: 0
Run terminated.
Run Summary
  Number of events processed : 100
  User=0.38000s Real=0.39000s Sys=0.02000s

============   SUMMARY   ============

   100 Events were generated
   Incident particle pi+ Ekin(meV) = 1000.00000

    Average number of secondaries

             World         Vertex Detector       Muon Detector

  Gammas:    7.04000        0.14000        0.70000
  Electrons: 13.39000        4.25000        3.30000
  Positrons:  0.41000        0.00000        3.30000

    Normalised energy deposition in the calorimeter

  E1 = 0.21979 +- 0.00146 RMS(E1) = 0.01463 +- 0.00146
  E9 = 0.22098 +- 0.00158 RMS(E9) = 0.01580 +- 0.00158
  E25= 0.22117 +- 0.00160 RMS(E25)= 0.01602 +- 0.00160

    Energy deposition in the absorbers

  Eabs1(MeV)= 0.78666 +- 0.02063 RMS(Eabs1)= 0.20626 +- 0.02063
  Eabs2(MeV)= 0.89956 +- 0.05071 RMS(Eabs2)= 0.50713 +- 0.05071
  Eabs3(MeV)= 0.88869 +- 0.03458 RMS(Eabs3)= 0.34576 +- 0.03458
  Eabs4(MeV)= 0.79623 +- 0.02766 RMS(Eabs4)= 0.27661 +- 0.02766

    Average number of hits in the vertex detector

  Nhit= 3.05000 +- 0.02598

--------- Ranecu engine status ---------
 Initial seed (index) = 0
 Current couple of seeds = 630634777, 290751365
----------------------------------------
#
/gun/particle ion
/gun/ion      6 12 6 0
/gun/energy 1 GeV
/run/beamOn 100
### Run 2 start  beam E= 1000.00000 MeV

--------- Ranecu engine status ---------
 Initial seed (index) = 0
 Current couple of seeds = 630634777, 290751365
----------------------------------------
Start Run processing.

---> Begin of Event: 0
Run terminated.
Run Summary
  Number of events processed : 100
  User=14.39000s Real=14.47000s Sys=0.03000s

============   SUMMARY   ============

   100 Events were generated
   Incident particle C12[0.0] Ekin(meV) = 1000.00000

    Average number of secondaries

             World         Vertex Detector       Muon Detector

  Gammas:    0.01000        2.39000        0.00000
  Electrons: 2.70000        551.12000        0.00000
  Positrons:  0.00000        0.00000        0.00000

    Normalised energy deposition in the calorimeter

  E1 = 0.68950 +- 0.00019 RMS(E1) = 0.00193 +- 0.00019
  E9 = 0.68950 +- 0.00019 RMS(E9) = 0.00193 +- 0.00019
  E25= 0.68950 +- 0.00019 RMS(E25)= 0.00193 +- 0.00019

    Energy deposition in the absorbers

  Eabs1(MeV)= 135.69962 +- 0.12610 RMS(Eabs1)= 1.26103 +- 0.12610
  Eabs2(MeV)= 154.15914 +- 0.12731 RMS(Eabs2)= 1.27310 +- 0.12731
  Eabs3(MeV)= 0.00000 +- 0.00000 RMS(Eabs3)= 0.00000 +- 0.00000
  Eabs4(MeV)= 0.00000 +- 0.00000 RMS(Eabs4)= 0.00000 +- 0.00000

    Average number of hits in the vertex detector

  Nhit= 14.11000 +- 0.34667

--------- Ranecu engine status ---------
 Initial seed (index) = 0
 Current couple of seeds = 611494742, 455036980
----------------------------------------
#
Graphics systems deleted.
Visualization Manager deleting...
G4 kernel has come to Quit state.
